src/DataStructures.jl

```diff
@@ -14,11 +14,11 @@ module DataStructures
                  ReverseOrdering, Reverse, Lt,
                  isless,
                  union, intersect, symdiff, setdiff, issubset,
-                 find, searchsortedfirst, searchsortedlast, endof, in
+                 find, searchsortedfirst, searchsortedlast, endof, in, produce
 
 
     export Deque, Stack, Queue
-    export deque, enqueue!, dequeue!, update!,iter
+    export deque, enqueue!, dequeue!, update!, forward_iter, backward_iter
```

Comment:
actually I like the idea of leaving iter but having it return the default iterator for the type